UnleashClient/__init__.py

```diff
@@ -450,69 +432,51 @@ def get_variant(self, feature_name: str, context: Optional[dict] = None) -> dict
         context = context or {}
         context.update(self.unleash_static_context)
 
-        if self.unleash_bootstrapped or self.is_initialized:
-            try:
-                feature = self.features[feature_name]
-
-                if not self._dependencies_are_satisfied(feature_name, context):
-                    feature.increment_stats(False)
-                    feature._count_variant("disabled")
-                    return DISABLED_VARIATION
-
-                variant_check = feature.get_variant(context)
-
-                if self.unleash_event_callback and feature.impression_data:
-                    try:
-                        event = UnleashEvent(
-                            event_type=UnleashEventType.VARIANT,
-                            event_id=uuid.uuid4(),
-                            context=context,
-                            enabled=variant_check["enabled"],
-                            feature_name=feature_name,
-                            variant=variant_check["name"],
-                        )
-
-                        self.unleash_event_callback(event)
-                    except Exception as excep:
-                        LOGGER.log(
-                            self.unleash_verbose_log_level,
-                            "Error in event callback: %s",
-                            excep,
-                        )
-                        return variant_check
-
-                return variant_check
-            except Exception as excep:
+        variant = self._resolve_variant(feature_name, context)
+
+        if not variant:
+            if self.unleash_bootstrapped or self.is_initialized:
                 LOGGER.log(
                     self.unleash_verbose_log_level,
-                    "Returning default flag/variation for feature: %s",
+                    "Attempted to get feature flag/variation %s, but client wasn't initialized!",
                     feature_name,
                 )
+            variant = DISABLED_VARIATION
```

Comment:
Would probably be cleaner to move this part over to `_resolve_variant` and make it always return a variant instead of either a variant or `None`?

Reply:
I left it alone because I was going to implement the fallback function, which needs to know if there's a variant or not... then I realised we never got around to implementing that. So I'm leaving this until we do actually implement it